src/state/opamFileTools.ml

```diff
@@ -1328,6 +1328,23 @@ let read_opam dir =
       (OpamPp.string_of_bad_format (OpamPp.Bad_format (snd err)));
     None
   | None, None -> None
+  | exception OpamPp.Bad_version ((_, _errmsg), Some version) ->
+    let sversion = OpamVersion.to_string version in
+    let scurrent = OpamVersion.to_string OpamVersion.current_nopatch in
+    log "opam-version %S unsupported on %s. Added as dummy unavailable package."
+      sversion (OpamFile.to_string opam_file);
+    Some
+      (OpamFile.OPAM.empty
+       |> OpamFile.OPAM.with_available
+         (FOp (FIdent ([], OpamVariable.of_string "opam-version", None),
+               `Geq, FString sversion))
+       |> OpamFile.OPAM.with_descr_body
+         (Printf.sprintf
+            "This package uses opam %s file format which opam %s cannot \
+             read.\n\n\
+             In order to install or view information on this package, please \
+             upgrade your opam installation to at least version %s."
```

Comment:
We should also redirect to a FAQ link, where we explain the issue and how to resolve it. It is easier to update the FAQ section with new "errors" (the futures ones that we try to handle _a priori_) than update the message in the package definition.
```suggestion
             upgrade your opam installation to at least version %s.\
             See https://opam.ocaml.org/doc/FAQ.html#Why-opam-tells-me-I-have-a-dummy-package."
```

Reply:
I'm not sure we want to do this here. This message will only appear in the description of the package and i think the rest of the message is descriptive enough.

However, future improvement could be made elsewhere in the code (somewhere more user-facing) as mentioned in the PR description:
> In the future it would be nice to have a specific error message to tell the user why this package is not available, but I feel like we can do this later (TODO: open a ticket for it if this is merged first)

Reply:
Discussed, no need to add it in FAQ